compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -880,17 +1438,41 @@ static Args parseArgs(int argc, char **argv) {
       ret.stdout = true;
     } else if (std::strcmp("--dump", argv[i]) == 0) {
       ret.dump = true;
+    } else if (std::strcmp("--text-only", argv[i]) ==  0) {
+      ret.textOnly = true;
+    } else if (std::strcmp("--output-dir", argv[i]) == 0) {
+      assert(i < (argc - 1));
+      ret.outputDir = argv[i + 1];
+      i += 1;
+    } else if (std::strcmp("--author", argv[i]) == 0) {
+      assert(i < (argc - 1));
+      ret.author = argv[i + 1];
+      i += 1;
+    } else if (std::strcmp("--process-used-modules", argv[i]) ==  0) {
+      ret.processUsedModules = true;
+    } else if (std::strcmp("--comment-style", argv[i]) == 0) {
+      assert(i < (argc - 1));
+      ret.commentStyle = argv[i + 1];
+      i += 1;
+    } else if (std::strcmp("--project-version", argv[i]) == 0) {
+      assert(i < (argc - 1));
+      ret.projectVersion = checkProjectVersion(argv[i + 1]);
+      i += 1;
     } else {
       ret.files.push_back(argv[i]);
     }
   }
   return ret;
 }
 
-// TODO what do we do if multiple modules are in one file?
+// TODO what do we do if multiple modules (submodules) are in one file?
 static std::string moduleName(const BuilderResult& builderResult) {
   for (const auto& ast : builderResult.topLevelExpressions()) {
     if (const Module* m = ast->toModule()) {
+      /* TODO: This is wrong if the module is an implicit module and it just
+          imports or uses another module. In that case the name should be that
+          of the used or imported module.
```

Comment:
It seems odd that chpldoc would care about use/imports? Did you mean if it's an implicit module that contains another module? If we added an implicit module, `m` would be the implicit module.

Reply:
I mean the pattern we see in `test/chpldoc/include/test-Include.doc.chpl`. I'm not sure it contains another module, but one file does the import, and the module it imports includes a submodule defined in a separate `.chpl` file.

Reply:
I guess chpldoc is trying to document all use/imported modules, then? I forget exactly what it's behavior with that is...

Reply:
there is a `chpldoc` flag for that `--process-used-modules`